service/src/test/java/uk/gov/mca/beacons/api/legacybeacon/application/LegacyBeaconServiceUnitTest.java

```diff
@@ -198,4 +201,92 @@ void delete_shouldUpdateTheNestedBeaconJsonObjectsLastModifiedDateToTodaysDate()
 
     Assertions.assertEquals(todaysDate, lastModifiedDate);
   }
+
+  @Test
+  void findHexIdsWithDuplicates_shouldOnlyReturnHexIdsThatMatchMoreThanOneBeacon()
+    throws Exception {
+    // arrange
+    LegacyBeacon legacyBeacon1 = LegacyBeaconTestUtils.initLegacyBeacon();
+    LegacyBeacon legacyBeacon2 = LegacyBeaconTestUtils.initLegacyBeacon();
+
+    legacyBeacon2.setHexId("666");
+
+    LegacyBeacon legacyBeaconWithUniqueHexId = LegacyBeaconTestUtils.initLegacyBeacon();
+    legacyBeaconWithUniqueHexId.setHexId("1D0EA08C52FFBFF");
+
+    LegacyBeacon legacyBeaconWithOtherUniqueHexId = LegacyBeaconTestUtils.initLegacyBeacon();
+    legacyBeaconWithOtherUniqueHexId.setHexId("1D0EA08C52FFBGG");
+
+    List<LegacyBeacon> legacyBeaconsInRepo = new ArrayList<>();
+    // 9D0E1D1B8C00001
+
+    when(mockLegacyBeaconRepository.findByHexIdNotNull())
+      .thenReturn(
+        List.of(
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon1,
+          legacyBeacon2,
+          legacyBeacon2,
+          legacyBeaconWithUniqueHexId,
+          legacyBeaconWithOtherUniqueHexId
+        )
+      );
+
+    // act
+    Map<String, Integer> duplicateHexIds = legacyBeaconService.findHexIdsWithDuplicates(
+      20,
+      0
+    );
+    int numberOfDuplicatesForFirstHexId = duplicateHexIds
+      .values()
+      .stream()
+      .collect(Collectors.toList())
+      .get(0);
+
+    // assert
+    Assertions.assertEquals(2, duplicateHexIds.size());
```

Comment:
We left this with a need for pagination in the API endpoint that gets duplicate beacons. The story is currently in 'Potentials for Next SoW to order'